changed json parser to allow mapping files of arbitrary structure

mguent opened the pull request.

mguent committed:
changed json parser to allow mapping files of arbitrary structure

RoelantVos reviewed (commented):
Hi @mguent thanks for this. That is really great, so this means any object can be used as input long as there is a matching handlebars pattern. And it's a bit more forgiving to deserialise as well.

I've added a commit which reverts the TemplateSampleBasic.handlebars back to its original form (I suspect this was a copy error - but please let me know if not).

While having a look I also moved the HandeBarsHelper.cs into the DataWarehouseAutomation library so it can be reused by both the CLI as well as the examples project. In my version, the code was duplicated (blush).

Also updated the handlebars NuGet version across all projects. This led to some minor small fixes across the board to get the helper functions working again.

Kind regards,
Roelant

Commit:
- Reverted TemplateSampleBasic.handlebars
- Moved HandleBarsHelper.cs to DataWarehouseAutomation project
- Upgraded to handlebars.net 2.0.4
- Added freeform to test projects / examples

mguent commented:
Hi Roelant, great, i am glad you find it useful :-) My update of TemplateSampleBasic.handlebars was accidental.
I also noticed the duplicated Helpers File.. it's a good thing to only keep one, and also the package update. 

Best,
Martin
